### PR TITLE
pypodman: add options to handle ssh host keys

### PR DESCRIPTION
**Pull request by afbjorklund**
Some hosts are creating new host keys every boot

### CODE REVIEW

File: contrib/python/pypodman/pypodman/lib/podman_parser.py [hunk 1]
```diff
@@ -207,6 +211,15 @@ def reqattr(name, value):
             or os.path.expanduser('~{}/.ssh/id_dsa'.format(args.username))
         )   # yapf:disable
 
+        setattr(
+            args,
+            'ignore_hosts',
+            getattr(args, 'ignore_hosts')
+            or os.environ.get('IGNORE_HOSTS')
```
[the remark this answers is not in the document]
Comment (afbjorklund) [reply contrib/python/pypodman/pypodman/lib/podman_parser.py]: It's in the `--help` ? The manpage for pypodman.conf is pretty sparse, so should be ok...

> pypodman.conf is one or more configuration files for running the pypodman command. pypodman.conf is a TOML file with the stanza [default], with a map of option: value.

Most people _should_ verify host keys, but the feature is not available on the server (yet)

[the remark this answers is not in the document]
Comment (rhatdan) [reply contrib/python/pypodman/pypodman/lib/podman_parser.py]: I would prefer that the Environment variables be prefixed with PODMAN_. I know we have not done this consistently thoughout the code.    Although I think we should fix it.
PODMAN_IGNORE_HOSTS

**Comment by rhatdan** [comment]
Other then the environment variable this PR LGTM.

File: contrib/python/pypodman/pypodman/lib/podman_parser.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (afbjorklund) [reply contrib/python/pypodman/pypodman/lib/podman_parser.py]: The name "ignore hosts" is not very good either, but best I could come up with.

The actual ssh configuration is `StrictHostKeyChecking=no` (and no known_hosts)

**Comment by afbjorklund** [comment]
Interestingly, ssh host keys has been an "important security feature" upstream (machine) since 2015.
Yet nothing has happened yet.

It is not so easy to get those host keys _back_ from the initial install, even if the ssh keys goes _in_ now.
Doesn't mean it won't happen.

**Comment by afbjorklund** [comment]
Think I fixed it. The only downside is that now it needs a "known_hosts" file instead.

i.e. so that you can override the path of `~user/.ssh/known_hosts`, when needed...

**Comment by rhatdan** [comment]
@jwhonce Please make sure you approve and merge.

**Comment by afbjorklund** [comment]
Only remaining issue is that each new IP/port will add a new line in the `known_hosts` file.

But I suppose that is to be expected, don't really want to start to _edit_ the host / start of the line:
`[localhost]:46565 ecdsa-sha2-nistp256 AAAAE2VjZHNhLXNoYTItbmlzdHAyNTYAAAAIbmlzdHAyNTYAAABBBKj1zFDfV90GraOkuLe8tQBIjKGaW7akYqd+UYx25ewqMrhzz/CYwCYHDivWUk1KOHiAYT/EpMWGknyJ/MDKxmA=`

As long as the IP remains constant (and the SSH port doesn't change), it should be fine...

**Comment by rhatdan** [comment]
bot, retest this please

**Comment by rhatdan** [comment]
bot, retest this please

**Comment by afbjorklund** [comment]
Guess this has to move to the new https://github.com/containers/python-podman repo, once the move is finished. I stayed from using `--ignore-hosts` in config but `$PODMAN_IGNORE_HOSTS` in env is OK...

i.e. mostly because unknown command parameters is a hard error, but unused environment variables are fine. The only annoyance is that ssh will ask you to confirm the host keys. Guess that is survivable ?


``` console
The authenticity of host '[127.0.0.1]:39003 ([127.0.0.1]:39003)' can't be established.
ECDSA key fingerprint is SHA256:6H8SkZ+DjQmRGCdhj/77CjxQsbm/ZsJoJ80Yhi30V3k.
Are you sure you want to continue connecting (yes/no)? 
```
The host key does survive `restart`, so you should only get one such question per machine (VM).

The work for known_hosts isn't completed yet anyway, but that would be much better than "ignore".

**Comment by rhatdan** [comment]
This PR Needs to move to podman-python.

**Pull request closed without merging**